dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/ajax/CMSMaintenanceAjax.java

```diff
@@ -376,4 +428,293 @@ public void zipTempDirectoryToStream(OutputStream out) throws IOException {
+					_xstream.addDefaultImplementation(com.dotcms.repackage.net.sf.hibernate.collection.List.class, java.util.List.class);
+					_xstream.addDefaultImplementation(com.dotcms.repackage.net.sf.hibernate.collection.Map.class, java.util.Map.class);
+					Mapper mapper = _xstream.getMapper();
+					_xstream.registerConverter(new HibernateCollectionConverter(mapper));
+					_xstream.registerConverter(new HibernateMapConverter(mapper));
+				}
+
+				int i= 0;
+				int step = 1000;
+				int total =0;
+				java.text.NumberFormat formatter = new java.text.DecimalFormat("0000000000");
+				/* we will only export 10,000,000 items of any given type */
+				for(i=0;i < 10000000;i=i+step){
+
+					_dh = new HibernateUtil(clazz);
+					_dh.setFirstResult(i);
+					_dh.setMaxResults(step);
+
+					//This line was previously like;
+					//_dh.setQuery("from " + clazz.getName() + " order by 1,2");
+					//This caused a problem when the database is Oracle because Oracle causes problems when the results are ordered
+					//by an NCLOB field. In the case of dot_containers table, the second field, CODE, is an NCLOB field. Because of this,
+					//ordering is done only on the first field for the tables, which is INODE
+					if(com.dotmarketing.beans.Tree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent, child, relation_type");
+					}
+					else if(MultiTree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent1, parent2, child, relation_type");
+					}
+					else if(TagInode.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by inode, tag_id");
+					}
+					else if(TemplateVersionInfo.class.equals(clazz)){
+						_dh.setSQLQuery("SELECT {template_version_info.*} from template_version_info template_version_info, identifier where identifier.id = template_version_info.identifier order by template_version_info.identifier ");
+					}
+					else if(ContainerVersionInfo.class.equals(clazz)){
+						_dh.setSQLQuery("SELECT {container_version_info.*} from container_version_info container_version_info, identifier where identifier.id = container_version_info.identifier order by container_version_info.identifier ");
+					}
+					else if(LinkVersionInfo.class.equals(clazz)){
+						_dh.setSQLQuery("SELECT {link_version_info.*} from link_version_info link_version_info, identifier where identifier.id = link_version_info.identifier order by link_version_info.identifier ");
+					}
+					else if(CalendarReminder.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by user_id, event_id, send_date");
+					} else if(Identifier.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent_path, id");
+					} else {
+						_dh.setQuery("from " + clazz.getName() + " order by 1");
+
+					}
+
+					_list = _dh.list();
+					if(_list.size() ==0){
+						try {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)